### ISSUE TITLE
Deployment Failed at ARM Configuration

### ISSUE THREAD

**Issue by dbroussa** [OfficeDev/microsoft-teams-apps-company-communicator#1013]
Tried to deploy multiple times to Azure and I get an error when I tries to deploy the ARM module.  It appears to throw a 503 Error

Error occurred while deploying Azure resources.
ERROR: ARM template deployment error.
At C:\Documents\GitHub\microsoft-teams-apps-company-communicator\Deployment\deploy.ps1:530 char:17
+                 Throw $deploymentExceptionMessage
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (ERROR: ARM template deployment error.:String) [], RuntimeException
    + FullyQualifiedErrorId : ERROR: ARM template deployment error.

Full error from PowerShell was:
Deploying app services, Azure function, bot service, and other supporting resources... (this step can take over an hour)ERROR: {"status":"Failed","error":{"code":"DeploymentFailed","message":"At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-deployment-operations for usage details.","details":[{"code":"BadRequest","message":"{\r\n  \"Code\": \"BadRequest\",\r\n  \"Message\": \"There was a conflict. Conflict with existing ScmType: ExternalGit\",\r\n  \"Target\": null,\r\n  \"Details\": [\r\n    {\r\n      \"Message\": \"There was a conflict. Conflict with existing ScmType: ExternalGit\"\r\n    },\r\n    {\r\n      \"Code\": \"BadRequest\"\r\n    },\r\n    {\r\n      \"ErrorEntity\": {\r\n        \"ExtendedCode\": \"01020\",\r\n        \"MessageTemplate\": \"There was a conflict. {0}\",\r\n        \"Parameters\": [\r\n          \"Conflict with existing ScmType: ExternalGit\"\r\n        ],\r\n        \"Code\": \"BadRequest\",\r\n        \"Message\": \"There was a conflict. Conflict with existing ScmType: ExternalGit\"\r\n      }\r\n    }\r\n  ],\r\n  \"Innererror\": null\r\n}"},{"code":"BadRequest","message":"{\r\n  \"error\": {\r\n    \"code\": \"RoleAssignmentUpdateNotPermitted\",\r\n    \"message\": \"Tenant ID, application ID, principal ID, and scope are not allowed to be updated.\"\r\n  }\r\n}"},{"code":"BadRequest","message":"{\r\n  \"error\": {\r\n    \"code\": \"RoleAssignmentUpdateNotPermitted\",\r\n    \"message\": \"Tenant ID, application ID, principal ID, and scope are not allowed to be updated.\"\r\n  }\r\n}"},{"code":"BadRequest","message":"{\r\n  \"error\": {\r\n    \"code\": \"RoleAssignmentUpdateNotPermitted\",\r\n    \"message\": \"Tenant ID, application ID, principal ID, and scope are not allowed to be updated.\"\r\n  }\r\n}"},{"code":"BadRequest","message":"{\r\n  \"error\": {\r\n    \"code\": \"RoleAssignmentUpdateNotPermitted\",\r\n    \"message\": \"Tenant ID, application ID, principal ID, and scope are not allowed to be updated.\"\r\n  }\r\n}"},{"code":"BadRequest","message":"{\r\n  \"error\": {\r\n    \"code\": \"RoleAssignmentUpdateNotPermitted\",\r\n    \"message\": \"Tenant ID, application ID, principal ID, and scope are not allowed to be updated.\"\r\n  }\r\n}"},{"code":"BadRequest","message":"{\r\n  \"error\": {\r\n    \"code\": \"RoleAssignmentUpdateNotPermitted\",\r\n    \"message\": \"Tenant ID, application ID, principal ID, and scope are not allowed to be updated.\"\r\n  }\r\n}"},{"code":"BadRequest","message":"{\r\n  \"error\": {\r\n    \"code\": \"RoleAssignmentUpdateNotPermitted\",\r\n    \"message\": y
r\n  }\r\n}"}]}}

**Comment by gsv022** [comment]
Hi @dbroussa ,
Thanks for raising the query. Here are the observations from the logs.

- _For the RoleAssignmentUpdateNotPermitted issue_, please follow the link and update the GUID's accordingly.
- To reach failed resources, Go to azure portal-->Go to resource group(Where the deployment has been done)-->Select  Data-Function from list of resources-->Go to Deployment Center-->Click on Settings

<img width="592" alt="image" src="https://user-images.githubusercontent.com/98143492/220826419-783f50cb-7868-492e-b35d-9392364b4b97.png">

- If the resource is not pointing to intended repository and  showing empty. Please connect to the external Git and point to the official repository.
- Once the status turns to Success(active) in the logs section then good to proceed with remaining steps of deployment.
Note  Do the same for all failed resources. If you are still facing the issue, to unblock yourself please follow the manual deployment guide.

**Comment by gsv022** [comment]
@dbroussa ,Could you please let us know the status

**Comment by v-jaygupta** [comment]
Hi @dbroussa ,

Thank you for bringing this issue to our attention. In order to maintain consistency in our issue tracking system, we are taking steps to address any items that may have been overlooked and did not receive a response. If you are still experiencing the issue and require assistance, please do not hesitate to let us know, and we will be happy to discuss further. However, if no response is received, we will assume that no further action is required, and we will proceed to close the issue within 3 business days. Please note that you can always open a new issue or respond here if the issue resurfaces.
